bricks/net/http/impl/server.h

```diff
@@ -597,14 +605,32 @@ class GenericHTTPServerConnection final : public HTTPResponder {
 
       // The actual implementation of sending HTTP chunk data.
       template <typename T>
-      void SendImpl(T&& data) {
-        if (!data.empty()) {
+      void SendImpl(T&& data, ChunkFlush flush) {
+        if (!data.empty() || (flush == ChunkFlush::Flush && cache_size_)) {
           try {
-            connection_.BlockingWrite(strings::Printf("%lX", data.size()), true);
-            connection_.BlockingWrite(constants::kCRLF, true);
-            connection_.BlockingWrite(std::forward<T>(data), true);
-            // Force every chunk to be sent out by passing `false` as the second argument.
-            connection_.BlockingWrite(constants::kCRLF, false);
+            if (!data.empty()) {
+              const auto chunk_header = strings::Printf("%lX", data.size()) + constants::kCRLF;
+              const auto chunk_size = chunk_header.size() + data.size() + constants::kCRLFLength;
+              if (cache_size_ && (flush == ChunkFlush::Flush || chunk_size > CACHE_SIZE - cache_size_)) {
+                connection_.BlockingWrite(data_cache_, cache_size_, true);
+                cache_size_ = 0;
+              }
+              if (flush == ChunkFlush::Flush || chunk_size > cache_size_) {
```

Comment:
Shouldn't this be `... || chunk_size > CACHE_SIZE - cache_size_` as above?

Reply:
Good point, it's actually a bad mistake, which appeared in the aca9f24. Sure, the condition should look like `chunk_size > CACHE_SIZE`. There is no need to substract the `cache_size_`, because if the `chunk_size` was bigger than `CACHE_SIZE -
 cache_size`, but smaller than the `CACHE_SIZE`, it would be checked above and as a result the `cache_size_` became zero.